checker/src/org/checkerframework/checker/index/qual/NegativeIndexFor.java

```diff
@@ -0,0 +1,27 @@
+package org.checkerframework.checker.index.qual;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.JavaExpression;
+import org.checkerframework.framework.qual.SubtypeOf;
+
+/**
+ * The annotated expression is between -1 and -a.length - 1, inclusive, of each sequence a listed in
```

Comment:
I personally find ranges/intervals easier to understand if the lower bound is stated before the upper bound.  Also, for a variable name that is also a word, like `a`, wrapping it in `{@code ...}` is helpful.